test/mongodb.test.js

```diff
@@ -116,6 +117,31 @@ describe('mongodb connector', function() {
         done();
       });
     });
+
+    it('should skip connect phase (lazyConnect = true)', function(done) {
+      var ds = getDataSource({
+        host: 'localhost',
+        port: 4,
+        lazyConnect: true
+      });
+      initialize(ds, function(err) {
+        should.not.exist(err);
+        done();
+      })
+    });
+
+    it('should report connection error (lazyConnect = false)', function(done) {
+      var ds = getDataSource({
+        host: 'localhost',
+        port: 4,
+        lazyConnect: false
+      });
+      initialize(ds, function(err) {
+        (!!err).should.be.true;
+        err.message.should.match(/connect ECONNREFUSED 127.0.0.1:4/);
```

Comment:
This is rather fragile and may break in the future. Also not that `localhost` may get resolved into an IPv6 address.

I am proposing to relax the check to something like`err.message.should.contain('ECONNREFUSED')`